Python/pystate.c

```diff
@@ -810,7 +810,12 @@ new_threadstate(PyInterpreterState *interp)
 {
     PyThreadState *tstate;
     _PyRuntimeState *runtime = interp->runtime;
-
+    // Allocate eagerly without lock
+    PyThreadState *new_tstate = alloc_threadstate();
```

Comment:
It would be worth providing a little more detail here:

```suggestion
    // We don't need to allocate a thread state for the main interpreter
    // (the common case), but doing it later for the other case revealed a
    // reentrancy problem (deadlock).  So for now we always allocate before
    // taking the interpreters lock.  See GH-96071.
    PyThreadState *new_tstate = alloc_threadstate();
```